pywbem_mock/_wbemconnection_mock.py

```diff
@@ -3043,31 +3043,34 @@ def _fake_invokemethod(self, methodname, objectname, Params, **params):
                              local_only=False, include_qualifiers=True,
                              include_classorigin=True)
 
+        # Determine if method defined in classname defined in
+        # the classorigin of the method
         try:
-            target_class = cc.methods[methodname].class_origin
+            target_cln = cc.methods[methodname].class_origin
         except KeyError:
             raise CIMError(CIM_ERR_METHOD_NOT_FOUND, 'Method %s not found '
-                           'in %s class hiearchy' % (methodname,
-                                                     localobject.classname))
-
-        # test for target class in methods repo
+                           'in class %s class' % (methodname,
+                                                  localobject.classname))
+        if target_cln != cc.classname:
+            # TODO FUTURE: add method to repo that allows privileged users
+            # direct access so we don't have to go through _get_class and can
+            # test classes directly in repo
+            tcc = self._get_class(target_cln, namespace,
+                                  local_only=False, include_qualifiers=True,
+                                  include_classorigin=True)
+            if methodname not in tcc.methods:
+                raise CIMError(CIM_ERR_METHOD_NOT_FOUND, 'Method %s not found '
+                               'in class %s' % (methodname, target_cln))
+
+        # Test for target class in methods repo
         try:
-            methods = methodsrepo[target_class]
+            methods = methodsrepo[target_cln]
         except KeyError:
-            raise CIMError(CIM_ERR_NOT_FOUND, 'Class %s for Method %s in '
-                                              'namespace %s not '
-                                              'registered in methods '
-                                              'repository' %
+            raise CIMError(CIM_ERR_METHOD_NOT_FOUND,
+                           'Class %s for method %s in namespace %s not '
+                           'registered in methods repository' %
```

Comment:
Can this ever happen? I think it can only happen when the implementation is incorrect. If that is the case, this should be an assertion error, not a CIMError.

Reply:
Actually I want to leave it in for now. In particular because this is the one I am hitting in the wbem_server test .  I am not sure why but originally it failed at the method level but once we did the the change to Deepcopy the error moved over to the the class not being there. To be honest, I did not look at why so far.  In any case I would like to get this in as it is and if necessary write another issue to clean up those few lines of code

Reply:
ok